storage/command_queue.go

```diff
@@ -74,45 +74,56 @@ func (cq *CommandQueue) onEvicted(key, value interface{}) {
 	}
 }
 
-// GetWait initializes the supplied wait group with the number of
-// executing commands which overlap the specified key range. If end is
-// empty, end is set to start.Next(), meaning the command affects a
-// single key. The caller should call wg.Wait() to wait for
-// confirmation that all gating commands have completed or
-// failed. readOnly is true if the requester is a read-only command;
-// false for read-write.
-func (cq *CommandQueue) GetWait(start, end proto.Key, readOnly bool, wg *sync.WaitGroup) {
-	// This gives us a memory-efficient end key if end is empty.
-	if len(end) == 0 {
-		end = start.Next()
-		start = end[:len(start)]
-	}
-	for _, c := range cq.cache.GetOverlaps(start, end) {
-		c := c.Value.(*cmd)
-		// Only add to the wait group if one of the commands isn't read-only.
-		if !readOnly || !c.readOnly {
-			c.pending = append(c.pending, wg)
-			wg.Add(1)
+// GetWait initializes the supplied wait group with the number of executing
+// commands which overlap the specified key ranges. If an end key is empty, it
+// only affects the start key. The caller should call wg.Wait() to wait for
+// confirmation that all gating commands have completed or failed. readOnly is
```

Comment:
> readOnly is true if the requester is a read-only command; false for read-write

weird to have this logic in the caller. can we pass the requester in? it's trivial to check if it's readonly here.